include/WCSFoFRoutine.h

```diff
@@ -0,0 +1,170 @@
+#ifndef WCSFOF_MATCH_H
+#define WCSFOF_MATCH_H
+
+#include "FoF.h"
+#include "Astrometry.h"
+#include "FitsTable.h"
+#include "Units.h"
+#include "FitSubroutines.h"
+
+#include <vector>
+
+using namespace std;
+using namespace img;
+
+using astrometry::WCS_UNIT;
+
+/// Struct that will hold the info about each point that matcher (and subsequent programs)
+/// will need
+struct Point
+{
+  Point(double x_, double y_, long ext, long obj, long expo) : x(2),
+                                                               extensionNumber(ext),
+                                                               objectNumber(obj),
+                                                               exposureNumber(expo)
+  {
+    x[0] = x_;
+    x[1] = y_;
+  }
+  vector<double> x;
+  long extensionNumber; // "extension" is an individual input FITS bintable.
+  long objectNumber;    // gives the object number of this Point in its input catalog
+  long exposureNumber;  // Which exposure the catalog is from.
+  const vector<double> getX() const { return x; }
+};
+
+// *** Here is the typedef that says we'll be collecting & matching Points
+// *** using the Catalog template class in FoF.h.  Change this typedef if
+// *** a better 2d matching structure is desired!
+typedef fof::Catalog<Point, 2> PointCat;
+
+struct Field
+{
+  string name;
+  // Coordinate system that has lat=lon=0 at field center and does projection to use for matching
+  unique_ptr<astrometry::SphericalCoords> projection;
+  double extent;
+  double matchRadius;
+  // Map from affinity name to its catalog of matches:
+  typedef map<string, PointCat *> CatMap;
```

Comment:
I have implemented this but am running into some problems where it is implemented. I am probably missing something trivial: I changed the class method definition for `catalogFor` to ` unique_ptr<PointCat> catalogFor(const string &affinity) { ... `, but I am getting some "error: use of deleted function " compiler errors for this and a few other instances where `catalogs` is used. Do I want `catalogFor` to return something different now?

Reply:
"Use of deleted function" _probably_ means you need to move some variable assignments or function calls in `std::move` - if the deleted function is the copy constructor or assignment operator, this is the compiler saying, "I can't copy this `unique_ptr`".

Reply:
In this case I wouldn't want to do a `std::move`, since I want the map `catalogs` to stay intact. Would it be legitimate to have `catalogFor` return a reference to a `unique_ptr`, or is the only option to do a `std::move` to get the `unique_ptr`, then `std::move` it back in when I am done using it?